### ISSUE TITLE
Blazor WASM Net 6 Preview 4 Azure AD - There was an error trying to log you in: 'Cannot read property 'toLowerCase' of undefined'

### ISSUE THREAD

**Issue by aherrick** [dotnet/aspnetcore#33312]
I have a simple Blazor WASM running Net 6 Preview 4 that I setup using this guide:

https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/hosted-with-azure-active-directory?view=aspnetcore-6.0

The exact code works perfect with Net 5.0 (using corresponding Net 5.0 assemblies)

However when I upgrade to Net 6 I receive the following Authentication error:

![eod4H](https://user-images.githubusercontent.com/586671/120893101-5aff6d00-c5df-11eb-8d19-c33128f7dc22.png)

`There was an error trying to log you in: 'Cannot read property 'toLowerCase' of undefined'`

I can't seem to get anymore detail either. This is published on Azure App Service Self Contained publish.

The code can be found here: https://github.com/aherrick/BlazorWASM.Net6Prev4.AzureAD

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
Could you please share a repro, which is using a clean .NET 6 Preview 4 app (not upgraded from an earlier version). If that still reproduces, we'll take a look.

**Comment by aherrick** [comment]
@mkArtakMSFT yep its linked here. Thanks!

https://github.com/aherrick/BlazorWASM.Net6Prev4.AzureAD

**Comment by crbrz** [comment]
The issue is caused by assembly trimming done in published version (the property `LoginMode` in `MsalProviderOptions` gets removed). 

To workaround it exclude `Microsoft.Authentication.WebAssembly.Msal` from trimming by adding this to the client project file (tester.Client.csproj):
```
  <ItemGroup>
    <TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />
  </ItemGroup>
```

**Comment by aherrick** [comment]
@crbrz thanks! is there more to it? I add that line and then receive the following error when publishing. 
![6-24-2021 10-45-32 AM](https://user-images.githubusercontent.com/586671/123283417-60165480-d4d9-11eb-9139-9424108e4d84.png)

**Comment by crbrz** [comment]
Delete all `bin` and `obj` folders from the solution, delete all the files in destination folder on the server before publishing again.

**Comment by aherrick** [comment]
@crbrz that worked! thank you!

**Comment by ScottyLightnin** [comment]
Worked for me too, thank you.

**Comment by anthonychu** [comment]
Is it possible to add `<TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />` to the project template so folks don't run into this?

**Comment by DAlumbaugh** [comment]
@javiercn - I'm interpreting this to mean that we can't use Blazor and Azure B2C with the initial release of .net 6?  Can you confirm so I can show my client?

**Comment by bryanp-ramosoil** [comment]
This is still an issue in VS2022. I've looked around and have found no answer. Just using the base template has this issue. I've tried all ideas with no luck. Is there a change in what's needed on the Azure app registration side? 
Any help would be appreciated. 

![image](https://user-images.githubusercontent.com/53661721/142700220-acc047b3-548e-4f27-86b6-e0b459aa8feb.png)

Thanks!!

**Comment by bryanp-ramosoil** [comment]
FYI: The mentioned workaround doesn't work in the latest version.

<TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />

![2021-11-19 14_50_42-RamosOil Intranet](https://user-images.githubusercontent.com/53661721/142701793-c3c04f30-1e9e-4e3e-baa1-e03ce424274b.png)

**Comment by parko65** [comment]
> FYI: The mentioned workaround doesn't work in the latest version.
> 
> ![2021-11-19 14_50_42-RamosOil Intranet](https://user-images.githubusercontent.com/53661721/142701793-c3c04f30-1e9e-4e3e-baa1-e03ce424274b.png)

bryanp-ramosoil You haven't got VS2022 installed alongside vs2019 by any chance?

**Comment by tiputrading** [comment]
@parko65  I have the same problem, I tried all the solutions, 
![image](https://user-images.githubusercontent.com/59416738/142771254-5c7de574-afd8-405e-bc21-db3d61230d68.png)

**Comment by bryanp-ramosoil** [comment]
I uninstalled VS2019 prior to installing VS2022. I was wondering if I should reinstall VS2019. Based on your issue it looks as if that wouldn't work.

**Comment by Trapulo** [comment]
I have a project upgraded from 5.0 to 6.0. All ok in debug, but when the app is published I am not able to perform a login. The login controller does NOT redirect users to Azure, and reports only the "RemoteAuthenticatorView.LogInFailed" template.

I have NOT in console the reported errors (in fact I haven't any error, only the user authenticated requirement failure), but using the suggested workaround I was able to publish a working app.

**Comment by tiputrading** [comment]
I also upgraded the project from 5.0 to 6.0, and got error There was an error trying to log you in: 'Cannot read property 'toLowerCase' of undefined', so I created new project using msidentity template (and by using VS > New Project) and published again, still same error, I then tried suggestion by @crbrz still no fix.

**Comment by parko65** [comment]
@tiputrading I had same issue with 6.0, had another machine with VS2019 .Net5.0, built new Hosted BlazorWasm sln with MSIdentity using dotnet CLI - see sample in docs [link](https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/hosted-with-azure-active-directory?view=aspnetcore-6.0). Configured AAD as per docs and all works ok.

**Comment by VultureJD** [comment]
Any update on this? Using the 6.0 template and following the [instructions](https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/hosted-with-azure-active-directory?view=aspnetcore-6.0) and then the steps here, I'm still getting the integrity failing:
`Failed to find a valid digest in the 'integrity' attribute for resource 'https://xxxxxxxxxxxxxx.azurewebsites.net/_framework/Microsoft.Authentication.WebAssembly.Msal.dll' with computed SHA-256 integrity '7G9gFCp+PjD9I0hpityl8yAr1zyzNL+tdrS3MpMHvl4='. The resource has been blocked.`

**Comment by parko65** [comment]
> Any update on this? Using the 6.0 template and following the [instructions](https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/hosted-with-azure-active-directory?view=aspnetcore-6.0) and then the steps here, I'm still getting the integrity failing: `Failed to find a valid digest in the 'integrity' attribute for resource 'https://xxxxxxxxxxxxxx.azurewebsites.net/_framework/Microsoft.Authentication.WebAssembly.Msal.dll' with computed SHA-256 integrity '7G9gFCp+PjD9I0hpityl8yAr1zyzNL+tdrS3MpMHvl4='. The resource has been blocked.`

Have you got VS2019 installed alongside VS2022?

**Comment by VultureJD** [comment]
@parko65 I do have both VS2019 and VS2022. I'm publishing directly from VS2022 into the App Service. If I publish in Debug, it all works fine, but Release throws the error above even when excluded from trimming.

**Comment by bryanp-ramosoil** [comment]
I tried publishing in debug to IIS and I'm still getting the following error.

There was an error trying to log you in: 'Cannot read properties of undefined (reading 'toLowerCase')'

This is getting frustrating

**Comment by parko65** [comment]
> I tried publishing in debug to IIS and I'm still getting the following error.
> 
> There was an error trying to log you in: 'Cannot read properties of undefined (reading 'toLowerCase')'
> 
> This is getting frustrating

@bryanp-ramosoil Are you publishing from VS2022? I had this problem also when both versions of VS were installed.

**Comment by bryanp-ramosoil** [comment]
@parko65  I've tried publishing in both VS2022 and VSCode. Results are the same. In both cases I published with the debug profile. 

Used this in VSCode : dotnet publish -c Debug -o ./publish

**Comment by Mr-Technician** [comment]
I just ran into this issue and the below fixed it. I'm running the latest VS 2022 and the latest .NET 6.

> Delete all `bin` and `obj` folders from the solution, delete all the files in destination folder on the server before publishing again.

**Comment by parko65** [comment]
@bryanp-ramosoil replicated your issue, created new blazorwasm using CLI using [https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/hosted-with-azure-active-directory?view=aspnetcore-6.0](url). Used @crbrz recommendations re trimming and deleting obj and bin subdirectories. Re published and all ok.

**Comment by bryanp-ramosoil** [comment]
@Mr-Technician and @parko65 It worked!!! Thanks!!!!

**Comment by VultureJD** [comment]
@parko65  @Mr-Technician Updating to VS2022 17.0.2 has fixed the issue. Can now publish in Release mode and it doesn't throw the integrity issue. Thanks!

**Comment by chgong** [comment]
The issue still exists even after I upgraded to VS 2022 17.0.2.
Applied the workaround suggested by @crbrz, the issue is gone.

**Comment by preservedmoose** [comment]
Same for me... I added this to the Blazor Client application settings
  <ItemGroup>
    <TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />
  </ItemGroup>
Then deleted all of the bin and obj folders in the solution.
I then stopped the site in Azure and published the solution.
I did not need to delete any folders on Azure...
After the publish completed, I started the site again and it is now working.
I had previously tried a Clean Solution / Rebuild Solution but it did not work - you have to manually delete the bin and obj folders.

**Comment by FrankAuwers** [comment]
I don't have this problem with .NET 6 on my dev machine, but when I deploy it to an Azure App Service, I get the same error.

**Comment by zzollner** [comment]
The location `<TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />` is added in the .csproj file does matter. 

Adding it to the end does not work. Placing it directly after your PackageReference item group seems to work.

**Comment by FrankAuwers** [comment]
I have alredy did it this way, but when it is hosted on Azure  I will receive the same error.

<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">

                <PropertyGroup>
                                <TargetFramework>net6.0</TargetFramework>
                                <Nullable>enable</Nullable>
                                <ImplicitUsings>enable</ImplicitUsings>
                                <ServiceWorkerAssetsManifest>service-worker-assets.js</ServiceWorkerAssetsManifest>
                </PropertyGroup>

                <ItemGroup>
                                <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="6.0.0" />
                                <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="6.0.0" PrivateAssets="all" />
                                <PackageReference Include="Microsoft.Authentication.WebAssembly.Msal" Version="6.0.0" />
                                <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
                                <PackageReference Include="Polly" Version="7.2.2" />
                                <PackageReference Include="Radzen.Blazor" Version="3.13.5" />
                </ItemGroup>

                <ItemGroup>
                                <TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />
                </ItemGroup>


Frank Auwers
***@***.***
 l www.scapta.com l
 +32 11 600 600 
Corda Campus l Kempische Steenweg 293 bus 38 l 3500 Hasselt 
This email may contain confidential information and/or copyright material. This email is intended for the use of the addressee only. Any unauthorized use may be unlawful. If you receive this mail by mistake, please advise the sender immediately by using the reply facility in your email software. Please don't print this e-mail unless you really need to.
From: Zach Zollner ***@***.***>
Sent: donderdag 9 december 2021 14:04
To: dotnet/aspnetcore ***@***.***>
Cc: Frank Auwers ***@***.***>; Comment ***@***.***>
Subject: [EXT]Re: [dotnet/aspnetcore] Blazor WASM Net 6 Preview 4 Azure AD - There was an error trying to log you in: 'Cannot read property 'toLowerCase' of undefined' (#33312)


Caution: Message from outside your organisation


The location <TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" /> is added in the .csproj file does matter.

Adding it to the end does not work. Placing it directly after your PackageReference item group seems to work.

-
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Faspnetcore%2Fissues%2F33312%23issuecomment-989833353&data=04%7C01%7CFrank.Auwers%40scapta.com%7C5dccf5fcb6504862658108d9bb145bc4%7C5d222bb8bedf40bd89de7511e02ce026%7C0%7C0%7C637746518385652560%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=psoJ5TwEahN3WZ5nJ7e3lHw3Kqpa9unoKGAQVkeRaQo%3D&reserved=0>, or unsubscribe<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAI6PKXGKHRQI4ID4YLFGGQTUQCSLVANCNFSM46ERM5UA&data=04%7C01%7CFrank.Auwers%40scapta.com%7C5dccf5fcb6504862658108d9bb145bc4%7C5d222bb8bedf40bd89de7511e02ce026%7C0%7C0%7C637746518385652560%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=nCkCoybgpWUjcA0v6GEJqpbxEeFMaNndD9fhbMQQFsc%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7CFrank.Auwers%40scapta.com%7C5dccf5fcb6504862658108d9bb145bc4%7C5d222bb8bedf40bd89de7511e02ce026%7C0%7C0%7C637746518385652560%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Slacxk2Nv6GRF7SX8YDtfsNGGmoUJjvFbox8KaTazRM%3D&reserved=0> or Android<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7CFrank.Auwers%40scapta.com%7C5dccf5fcb6504862658108d9bb145bc4%7C5d222bb8bedf40bd89de7511e02ce026%7C0%7C0%7C637746518385652560%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=rXQLciMGQnuhH8P3RNtydwDKjOC%2FZRAR5euNGehvey8%3D&reserved=0>.

**Comment by dstretch** [comment]
I was performing the build on an AzD build server (on-premises) and had to RDP to that and delete the obj and bin folders from the temporary build directory to resolve this error.

Path for me was: `<Installation folder of agent>\_work\<numbered build folder e.g. 165>\s\<AzD Project name>\<C# Project name>\obj`

Hope that helps someone else.

**Comment by mkb13** [comment]
> Same for me... I added this to the Blazor Client application settings
> 
> ![image](https://user-images.githubusercontent.com/26547539/145280081-1fc21e37-3491-42ac-9b28-32866d85bf81.png)
> 
> Then deleted all of the **bin** and **obj** folders in the solution. I then stopped the site in Azure and published the solution. _I did not need to delete any folders on Azure..._ After the publish completed, I started the site again and it is now working. I had previously tried a Clean Solution / Rebuild Solution but it did not work - you have to manually delete the **bin** and **obj** folders.

How do you stop the site in Azure?  I am using Azure Static Web Apps and I am not sure how to access the files to delete them nor how to stop the site (which I guess wouldn't be stopped, technically, because it's static?).  I am publishing via github.  This problem is seriously annoying me -- I have had so many issues trying to get things to work in Blazor Web Assembly that don't seem to quite work right and it doesn't seem to be getting a ton of attention.

**Comment by preservedmoose** [comment]
@mkb13 go to App Services, select your application, Overview in the sidebar and you can stop and start the site there.
![image](https://user-images.githubusercontent.com/26547539/147279814-362cd682-2c30-4e64-8beb-5ae9864c16d7.png)

**Comment by dstretch** [comment]
They're using Static Web Sites, so no App Service. As long as you've
overwritten the files it should be fine.

The bits you need to delete (obj & bin folders) are on your build
server/machine.

**Comment by 130nk3r5** [comment]
So,
I had to do the whole thing below, on both client and server project...
`<ItemGroup>
    <TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />
  </ItemGroup>`

As well as this on manual publish
![image](https://user-images.githubusercontent.com/4056738/148363618-0a1388aa-a191-4a8c-9704-78e0397b0cb2.png)

**Comment by JonathanKrantz99** [comment]
I had the same issue when trying to sign in. Updating Microsoft.Identity.Web and Microsoft.Identity.Web.UI to latest version(1.22.1) and then deleting the bin and obj folders solved the problem for me.

**Comment by markusrt** [comment]
@crbrz Thanks for the proposed workaround. It fixed the issue on my release docker build. Your answer sounded also like you know a potential root cause.

@mkArtakMSFT is this still planned in for any future sprint?

**Comment by stephajn** [comment]
I was having this problem as well on .NET 6.0.1 using the latest version of the MSAL library. The only problem was that I wasn't even getting any kind of error message. I would only get an empty string. Adding in the TrimRootAssembly node in my project files resolved this issue for me in multiple projects.

This issue has been open since June of 2021 and as of this writing it is February 2022. Does this mean that everyone who runs .NET 6 Blazor WASM and tries to do a Release build is having this problem? Shouldn't something like this have been fixed before .NET 6 went to RTM? Even more so, shouldn't it have been prioritized for the 6.0.1 release?

Or will this become part of the standard project file template if you decide to use MSAL when creating a new project?

**Comment by 9swampy** [comment]
Visual Studio 17.1.0
New WebAssembly Blazor site with Windows Identity Provider option.
Unmodified except to adding config (works in Dev fine), added trim to csproj and updated all packages to latest:

```
<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">

	<PropertyGroup>
		<TargetFramework>net6.0</TargetFramework>
		<Nullable>enable</Nullable>
		<ImplicitUsings>disable</ImplicitUsings>
		<ServiceWorkerAssetsManifest>service-worker-assets.js</ServiceWorkerAssetsManifest>
	</PropertyGroup>

	<ItemGroup>
		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="6.0.2" />
		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="6.0.2" PrivateAssets="all" />
		<PackageReference Include="Microsoft.Authentication.WebAssembly.Msal" Version="6.0.2" />
	</ItemGroup>

	<ItemGroup>
		<TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />
	</ItemGroup>

	<ItemGroup>
		<ServiceWorker Include="wwwroot\service-worker.js" PublishedContent="wwwroot\service-worker.published.js" />
	</ItemGroup>

</Project>
```

Publish to folder, FTP to server; site runs but log-in fails with:
`There was an error trying to log you in: 'Cannot read properties of undefined (reading 'toLowerCase')'`

I've done the delete bin/obj and cleared target before deploy. Have I missed anything?

**Comment by Mason742** [comment]
I was having this problem as well on .NET 6.0.2 and VS2022. 
Publish to folder, FTP to server; site runs but log-in fails with:
`There was an error trying to log you in: 'Cannot read properties of undefined (reading 'toLowerCase')'`

Added this line, closed VS, and deleted my bin/obj folders then republished and it worked.
```xml
<ItemGroup> 
     <TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />
</ItemGroup>
```

**Comment by IvanFeric** [comment]
I ran into the same problem myself and wanted to create a bug with my temporary solution. And it was only now that I found that it has already been reported.

The temporary solution that I made is similar to the one reported here but a bit different. I also noticed that publish operation is trimming some of the MSIL-related classes. However, I didn't want to fully prevent the trimming for the entire `Microsoft.Authentication.WebAssembly.Msal` assembly but only for the involved classes.

So, I added this in the Blazor .csproj file:
```xml
<ItemGroup>
	<TrimmerRootDescriptor Include="TrimmerDescriptors.xml" />
</ItemGroup>
```
Additionally, I added the `TrimmerDescriptors.xml` file with the following content:
```xml
<?xml version="1.0" encoding="utf-8" ?>
<linker>
  <assembly fullname="Microsoft.Authentication.WebAssembly.Msal">
    <type fullname="Microsoft.Authentication.WebAssembly.Msal.Models.MsalProviderOptions" preserve="all" />
    <type fullname="Microsoft.Authentication.WebAssembly.Msal.Models.MsalCacheOptions" preserve="all" />
    <type fullname="Microsoft.Authentication.WebAssembly.Msal.MsalAuthenticationOptions" preserve="all" />
  </assembly>
</linker>
```
For me, only those 3 classes represented the problem so I forced the trimmer to fully preserve them while it can safely trim the rest of the assembly.

**Comment by lnaie** [comment]
I had the same problem while migrating from v 5.0 to 6.04 and this one fixed it. 
Thanks!

> I was having this problem as well on .NET 6.0.2 and VS2022. Publish to folder, FTP to server; site runs but log-in fails with: `There was an error trying to log you in: 'Cannot read properties of undefined (reading 'toLowerCase')'`
> 
> Added this line, closed VS, and deleted my bin/obj folders then republished and it worked.
> 
> ```
> <ItemGroup> 
>      <TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />
> </ItemGroup>
> ```

**Comment by jing8956** [comment]
> The issue is caused by assembly trimming done in published version (the property `LoginMode` in `MsalProviderOptions` gets removed). 

@crbrz Thanks, I added `DynamicDependencyAttribute` to the main function and it worked as well, proving you were right.
```csharp
static class Program
{
    [DynamicDependency(DynamicallyAccessedMemberTypes.PublicMethods, typeof(MsalProviderOptions))]
    static async Task Main(string[] args)
    {
        var builder = WebAssemblyHostBuilder.CreateDefault(args);

        builder.RootComponents.Add<App>("#app");
        builder.RootComponents.Add<HeadOutlet>("head::after");

        builder.Services.AddScoped(sp => new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });

        builder.Services.AddMsalAuthentication(options =>
        {
            builder.Configuration.Bind("AzureAd", options.ProviderOptions.Authentication);
        });

        await builder.Build().RunAsync();
    }
}
```

**Comment by jt-pt-dev** [comment]
I encountered this issue still today.

A new Blazor WASM .NET hosted solution using AAD B2C for authentication. Straight from the Visual Studio template (VS 17.3.0), built using GitHub Actions and deployed to Azure App Service. Works locally but not when deployed. Gives the error below when navigating to `/fetchdata`:

`There was an error trying to log you in: 'this._settings.loginMode is undefined'`

The *.csproj fix worked, but it would be nice not to have to modify an out of the box template that uses all the MS products (Git, B2C, App Service).

```
<ItemGroup>
    <TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />
</ItemGroup>
```

**Comment by Lemraj** [comment]
Hi @crbrz @mkArtakMSFT,
I am having the same issue and the workaround seems not to work for me. After applying the workaround, the wasm app hangs on "Checking login state..."
![image](https://user-images.githubusercontent.com/10677560/185020145-3fdf3e87-1f4b-468c-a4e9-8c156d9d5e1c.png)

Just to be sure that the issue is not related to my solution, i have created a new wasm app, added authentication and got exactly the same issue.

thanks in advance for any help

**Comment by Mr-Technician** [comment]
@Lemraj are there any errors in your console?

**Comment by stanbeamish** [comment]
Hi,
we had the same issue and none of the above really helped.
But found a solution (that included the above modification in `*.csproj`. Maybe this helps someone else:

In `index.html` we have shuffled the order of the  `<script>`-tags at the end of the `<body>`-tag. I.e. the 
`<script src="_framework/blazor.webassembly.js" autostart="false"></script>`
is now first.
Then comes
`<script src="_content/Microsoft.Authentication.WebAssembly.Msal/AuthenticationService.js"></script>`.
Hereafter your other js-Files needed.
Lastly any PWA Configuration js-Files (that we need).
If necessary for your project, this moved in the `<head>`:
`<script src="_content/[..your..application..]/JsInterop.js"></script>`

This worked for us and champagne is now in the glasses

**Comment by Lemraj** [comment]
@Mr-Technician the console is clean And No errors PR warnings are logged.

@stanbeamish thanks for the suggestion. I will try this as soon as arrive home and let you know if it worked for me.

**Comment by Lemraj** [comment]
@stanbeamish unfortunately your suggestion did not solve my problem :(

Can anyone help me please with this issue?

**Comment by javiercn** [comment]
We believe this has been addressed in 7.0 as part of https://github.com/dotnet/aspnetcore/pull/43954.

The fix will be available in RC2.
